### PR TITLE
unigine-sanctuary: init at 2.3

### PR DESCRIPTION
**Pull request by barinov274**
###### Description of changes

I want to add another unigine benchmark package. It's unigine-sanctuary. It's a benchmarking tool from unigine. Nixos users will be able to install this package on their computers and benchmark gpu.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin

### CODE REVIEW

File: pkgs/applications/graphics/unigine-sanctuary/default.nix [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    install -D -m 0755 $name/1024x768_windowed.sh $out/bin/Sanctuary
+
+    cp -R $name/data $out/lib/unigine/sanctuary
+
+    wrapProgram $out/bin/Sanctuary \
+      --prefix LD_LIBRARY_PATH : $libPath:$out/lib/unigine/sanctuary/bin \
+      --run "cd $out/lib/unigine/sanctuary"
+
+    convert -size 256x256 xc:Transparent -define gradient:center="128,128" -define gradient:vector="128,128 128,128" -define gradient:radii="128,128" -fill radial-gradient:'rgb(164,0,0)-rgb(67,1,3)' -draw "roundRectangle 0,0 256,256 50,50" $name/icon.png
+    convert $name/icon.png -size 181.991x181.991 -font ${liberation_ttf}/share/fonts/truetype/LiberationSans-Regular.ttf -pointsize 181.991 -define gradient:center="128,128" -define gradient:vector="128,128 128,128" -define gradient:radii="46.6974,46.6974" -fill radial-gradient:'rgb(249,197,46)-rgb(218,144,31)' -stroke none -strokewidth 4.54977 -draw 'text 69.3061,194.247 "S"' $name/icon.png
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/unigine-sanctuary/default.nix:50]:
```suggestion
    convert -size 256x256 xc:Transparent -define gradient:center="128,128" -define gradient:vector="128,128 128,128" \
      -define gradient:radii="128,128" -fill radial-gradient:'rgb(164,0,0)-rgb(67,1,3)' \
      -draw "roundRectangle 0,0 256,256 50,50" $name/icon.png
    convert ${pname}-${version}/icon.png -size 181.991x181.991 -font ${liberation_ttf}/share/fonts/truetype/LiberationSans-Regular.ttf \
      -pointsize 181.991 -define gradient:center="128,128" -define gradient:vector="128,128 128,128" \
      -define gradient:radii="46.6974,46.6974" -fill radial-gradient:'rgb(249,197,46)-rgb(218,144,31)' \
      -stroke none -strokewidth 4.54977 -draw 'text 69.3061,194.247 "S"' ${pname}-${version}/icon.png
```

File: pkgs/applications/graphics/unigine-sanctuary/default.nix [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  ];
+
+  installPhase = ''
+    bash $src --target $name
+
+    install -D -m 0755 $name/bin/libUnigine_x86.so $out/lib/unigine/sanctuary/bin/libUnigine_x86.so
+    install -D -m 0755 $name/bin/Sanctuary $out/lib/unigine/sanctuary/bin/Sanctuary
+    install -D -m 0755 $name/1024x768_windowed.sh $out/bin/Sanctuary
+
+    cp -R $name/data $out/lib/unigine/sanctuary
```
[the remark this answers is not in the document]
Comment (barinov274) [reply pkgs/applications/graphics/unigine-sanctuary/default.nix]: error: undefined variable 'pname'
error: undefined variable 'verison'

it works without braces '{}'

File: pkgs/applications/graphics/unigine-sanctuary/default.nix [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+    wrapProgram $out/bin/Sanctuary \
+      --prefix LD_LIBRARY_PATH : $libPath:$out/lib/unigine/sanctuary/bin \
```
[the remark this answers is not in the document]
Comment (barinov274) [reply pkgs/applications/graphics/unigine-sanctuary/default.nix:46]: error: undefined variable 'libPath'
it also works only without braces '{}'

[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/applications/graphics/unigine-sanctuary/default.nix:46]: You are missing a rec next to mkDerivation.

[the remark this answers is not in the document]
Comment (barinov274) [reply pkgs/applications/graphics/unigine-sanctuary/default.nix]: ok it works with rec

File: pkgs/applications/graphics/unigine-sanctuary/default.nix [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    convert ${pname}-${version}/icon.png -size 181.991x181.991 -font ${liberation_ttf}/share/fonts/truetype/LiberationSans-Regular.ttf \
+      -pointsize 181.991 -define gradient:center="128,128" -define gradient:vector="128,128 128,128" \
+      -define gradient:radii="46.6974,46.6974" -fill radial-gradient:'rgb(249,197,46)-rgb(218,144,31)' \
+      -stroke none -strokewidth 4.54977 -draw 'text 69.3061,194.247 "S"' ${pname}-${version}/icon.png
+
+    for RES in 16 24 32 48 64 128 256; do
+      mkdir -p $out/share/icons/hicolor/"$RES"x"$RES"/apps
+      convert $name/icon.png -resize "$RES"x"$RES" $out/share/icons/hicolor/"$RES"x"$RES"/apps/Sanctuary.png
+    done
+    convert $name/icon.png -resize 128x128 $out/share/icons/Sanctuary.png
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/unigine-sanctuary/default.nix:58]:
```suggestion
      convert ${pname}-${version}/icon.png -resize "$RES"x"$RES" $out/share/icons/hicolor/"$RES"x"$RES"/apps/Sanctuary.png
    done
    convert ${pname}-${version}/icon.png -resize 128x128 $out/share/icons/Sanctuary.png
```

File: pkgs/applications/graphics/unigine-sanctuary/default.nix [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+  };
+
+  libPath = lib.makeLibraryPath [
+    libglvnd
+    openal
+    glibc
+  ];
+
+  installPhase = ''
+    bash $src --target $name
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/unigine-sanctuary/default.nix:34]:
```suggestion
    bash $src --target ${pname}-${version}
```

File: pkgs/applications/graphics/unigine-sanctuary/default.nix [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+    install -D -m 0755 $name/bin/libUnigine_x86.so $out/lib/unigine/sanctuary/bin/libUnigine_x86.so
+    install -D -m 0755 $name/bin/Sanctuary $out/lib/unigine/sanctuary/bin/Sanctuary
+    install -D -m 0755 $name/1024x768_windowed.sh $out/bin/Sanctuary
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/unigine-sanctuary/default.nix:38]:
```suggestion
    install -D -m 0755 ${pname}-${version}/bin/libUnigine_x86.so $out/lib/unigine/sanctuary/bin/libUnigine_x86.so
    install -D -m 0755 ${pname}-${version}/bin/Sanctuary $out/lib/unigine/sanctuary/bin/Sanctuary
    install -D -m 0755 ${pname}-${version}/1024x768_windowed.sh $out/bin/Sanctuary
```

File: pkgs/applications/graphics/unigine-sanctuary/default.nix [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+, copyDesktopItems
+, imagemagick
+, liberation_ttf
+}:
+stdenv.mkDerivation rec{
+  pname = "unigine-sanctuary";
+  version = "2.3";
+
+  src = fetchurl {
+    url = "http://m12-assets.unigine.com/d/Unigine_Sanctuary-${version}.run";
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/unigine-sanctuary/default.nix:23]:
```suggestion
    url = "https://m12-assets.unigine.com/d/Unigine_Sanctuary-${version}.run";
```

**Pull request merged**